mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -443,12 +425,11 @@ private[ml] trait GBTParams extends TreeEnsembleParams with HasMaxIter with HasS
    * (default = 0.1)
    * @group setParam
    */
-  def setStepSize(value: Double): this.type = set(stepSize, value)
-
-  override def validateParams(): Unit = {
+  def setStepSize(value: Double): this.type = {
     require(ParamValidators.inRange(0, 1, lowerInclusive = false, upperInclusive = true)(
```

Comment:
This validation should really be in the Param itself.  It's not since it's a sharedParam, but perhaps we should just copy the shared Param code here.  The problem with putting validation in the setter is that Params can get set in other ways too (such as ```gbt.set(stepSize, value)```